research-software-directory/RSD-as-a-service
# fix: encode search term to support 'reserved' characters like + and others

# Fix search on software, project and organisation page

Changes proposed in this pull request:
* encode search param to allow use of "reserved" characters like +, { and others      

How to test:
* `make start` to build and generate test data.
* try searching for C++ on software overview page. The url param should be encoded and C++ should be visible in search box.
* try searching for C++ on project overview page. The url param should be encoded and C++ should be visible in search box.
* try searching for C++ on organisation page. The url param should be encoded and C++ should be visible in search box.

## Example C++ search for software

![image](https://github.com/user-attachments/assets/83df173e-be85-4920-b4fa-c7efbea3c857)


PR Checklist:
